File: src/main/kotlin/me/icebear03/splendidenchants/api/EnchantAPI.java
package me.icebear03.splendidenchants.api;

import me.icebear03.splendidenchants.enchant.SplendidEnchant;
import org.bukkit.enchantments.Enchantment;

public class EnchantAPI {
    public static SplendidEnchant getSplendidEnchant(String idOrName) {
        return null;
        //TODO 不支持带颜色的name
    }

    public static SplendidEnchant getSplendidEnchant(Enchantment enchant) {
        return getSplendidEnchant(enchant.getKey().getKey());
    }

    public static String getName(Enchantment enchant) {
        return getSplendidEnchant(enchant).basicData.getName();
    }
}

File: src/main/kotlin/me/icebear03/splendidenchants/api/EnchantAPI.kt
package me.icebear03.splendidenchants.api

import me.icebear03.splendidenchants.enchant.EnchantLoader
import me.icebear03.splendidenchants.enchant.SplendidEnchant
import org.bukkit.enchantments.Enchantment

object EnchantAPI {

    fun getSplendidEnchant(idOrName: String): SplendidEnchant? {
        return EnchantLoader.enchantById[idOrName] ?: EnchantLoader.enchantByName[idOrName]
    }

    fun getSplendidEnchant(enchant: Enchantment): SplendidEnchant {
        return getSplendidEnchant(enchant.key.key)!!
    }

    fun getName(enchant: Enchantment): String {
        return getSplendidEnchant(enchant).basicData.name
    }

    fun getId(enchant: Enchantment): String {
        return getSplendidEnchant(enchant).basicData.id
    }

    fun isSame(a: Enchantment, b: Enchantment): Boolean = getId(a) == getId(b)
}